app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -123,8 +123,9 @@ echo "#"
 register_katello_host(){
     UUID=$(subscription-manager identity | head -1 | awk '{print $3}')
     curl --silent --show-error --cacert $KATELLO_SERVER_CA_CERT --request POST "<%= @registration_url %>" \
-         --data "uuid=$UUID" \
          <%= headers.join(' ') %> \
+         --data "uuid=$UUID" \
+         --data "host[build]=false" \
```

Comment:
I'm tempted to say that a host should by default be created as an unmanaged host from the registration workflow, in which case build mode doesn't matter.

Reply:
> I'm tempted to say that a host should by default be created as an unmanaged host from the registration workflow

Usually they are, but when you use registration for provisioned hosts, then they are managed.

Reply:
I'm not quite sure I follow. In my mind how it works is that before you register a host then there is no Host DB entry. After registration there's an unmanaged Host DB entry. If the user want to provision it, they have to later make it a managed host.

Reply:
> I'm not quite sure I follow. In my mind how it works is that before you register a host then there is no Host DB entry. 

Not in this case, because the host record is already there, from the provisioning.
Users first provision the host, then use global registration for the subscription configuration and other stuff happening in host_init_config template.

Or in different case it doesn't have to be right after the provisioning, but later when they want to re-register the host to a different capsule. With the current state, it will also re-run all assigned ansible roles, which is not desired behavior.

See the [BZ comment](https://bugzilla.redhat.com/show_bug.cgi?id=2212630#c13) for more info.

Reply:
I never thought of using registration to alter an existing host. To me registration implies something is always new. A good example of us needing to capture the scope of a feature.

Reply:
Yeah, same for me, but since we are deprecating bootstrap.py, users expect the GR to behave the same as the bootstrap.py, as @evgeni commented in the BZ.

Reply:
When we run GR, the host might be "new" (=unknown) to Foreman, or it might be an _existing_ one that users want to re-register (there are various reasons to do so, moving between content sources is one, moving to a different Lifecycle Environment/Content View is another -- yes that is used if the host owner does not have access to Katello directly, but knows the AK etc).
But it will be always an "existing" host in terms of provisioning, there is never a system to be PXE-booted, no OS to be installed etc in that flow.
Based on the above, I would say it's perfectly safe to *always* set `build=false` and let the rest flow as is.

Now, I know that some users *would* like to have the *post* provisioning steps to happen (= REX keys, Ansible roles, etc), and I also know that some others very much would not want that.

With that addition, I think we should:
* get this PR in as is, as it is changing the flow to be more "just register the host and have no surprises"
* add an *optional* flag for users so they can trigger post-provisioning ("finish"?) steps via GR

Reply:
I'd be mostly in favor of what @evgeni suggests. I think that the initial configuration is important in the regular (new host, unknown to Foreman) registration flow and I believe that's the majority of registration calls. But I can see the need of re-registering existing hosts where the post provisioning steps may not be desired.

IIRC setting the build mode to true had two good outcomes. One is, host build status is created and one can observe the host record in Foreman and could easily tell whether it's still being built (registered), it's switched to built only as the last step. Second, thanks to that, the build data can be saved, this would contain information about errors that appeared during the registration.

So I'd no set the build always to false, instead I'd agree the second part of the proposal (get the PR in as is, add the optional flag as a separate PR).

Reply:
> IIRC setting the build mode to true had two good outcomes. One is, host build status is created and one can observe the host record in Foreman and could easily tell whether it's still being built (registered), it's switched to built only as the last step. Second, thanks to that, the build data can be saved, this would contain information about errors that appeared during the registration.

This does give a risk of data loss. If the system is in build mode and reboots, it may end up reprovisioning which can destroy data on the systems. It feels like we're abusing one mechanism for another.

Reply:
@ares @ekohl so which path of fix are we going to choose?
I'm for merging it as it is right now and later adding a new optional flag for host-rebuild.

Reply:
My understanding is, the current behavior is, we always set the build=true (#setBuild). So while the concern about the data loss is valid, it's applicable to current version. This PR disables this for managed hosts, in other words, in the use case of using GR for existing hosts. That's why I'm in favor of merging as is to fix the reported issue for re-registration and we can continue debating whether we should also stop setting the build to true for newly registered hosts in a separate PR.

So I'm in favor of merging as is, as long as my understanding of the proposed change is correct.

Reply:
> I still fail to see why the controller can't default to build=false and only call setBuild if explicitly asked for.

@ekohl We use it for tracking how long it took to register the host (and also for tracking the state of registration)

> So I'm in favor of merging as is, as long as my understanding of the proposed change is correct.

I agree, it's for the upcoming downstream release, we should fix the issue ASAP and then debate over the usage of build during the registration